### PR TITLE
fix: accountstorage coverage, natspec, storage root derivation

### PR DESCRIPTION
**Pull request by adamegyed**
## Motivation

Cleanup pass on `AccountStorage.sol` and `AccountStorageInitializable.sol`.

## Solution

Expand and clarify natspec.

Test an edge case for the disableInitializers logic (impossible to reach in MA/SMA code).

Update the derivation of the storage root. Previously, we were not actually using ERC-7201, and the hash calculation was wrong. Both of those are now corrected, and the derivation matches the pattern used in MAv1, but with the major version number increased to 2.

### CODE REVIEW

File: src/account/AccountStorage.sol [hunk 1]
```diff
@@ -5,17 +5,18 @@ import {HookConfig, ModuleEntity} from "@erc6900/reference-implementation/interf
 
 import {LinkedListSet, SetValue} from "../libraries/LinkedListSetLib.sol";
 
-// bytes = keccak256("ERC6900.ModularAccount.Storage")
-bytes32 constant _ACCOUNT_STORAGE_SLOT = 0xc531f081ecdb5a90f38c197521797881a6e5c752a7d451780f325a95f8b91f45;
+// ERC-7201 derived storage slot.
+// keccak256(abi.encode(uint256(keccak256("Alchemy.ModularAccount.Storage_V2")) - 1)) & ~bytes32(uint256(0xff))
+bytes32 constant _ACCOUNT_STORAGE_SLOT = 0x596912a710dec01bac203cb0ed2c7e56a2ce6b2a68276967fff6dd57561bdd00;
 
 // Represents data associated with a specifc function selector.
```
Comment (jaypaik) [inline src/account/AccountStorage.sol]: Should we natspec this? I think foundry picks it up

**Review by jaypaik — APPROVED** [review]

File: src/account/AccountStorage.sol (hunk 1 above)
Comment (adamegyed) [reply src/account/AccountStorage.sol]: Yep, updating

Comment (jaypaik) [reply src/account/AccountStorage.sol]: Oh I meant this line (also a typo)
```suggestion
/// @dev Represents data associated with a specific function selector.
```

Comment (adamegyed) [reply src/account/AccountStorage.sol]: Fixed!
